src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -1064,29 +1064,17 @@ void Transformations::MainSnippets(void) {
                              ((in_type0 == element::f32 && in_type1 == ov::element::f32 &&
                                config.inferencePrecision == ov::element::bf16));
         const auto is_int8 = in_type0 == ov::element::i8;
-        if (is_fp32)
-            return true;
-        // Only FP32 dynamic MHA is supported
-        if (matmul->is_dynamic())
-            return false;
         if (matmul->get_transpose_a())
             return false;
-        // [150842] The execution of Brgemm INT8/BF16/FP16 on AMX platforms depends on the value of "K % VNNIFactor".
-        //          For more details, please teake a look at the ticket 150842
-        if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_amx)) {
-            const auto& b_shape = matmul->get_input_partial_shape(1);
-            const auto K = matmul->get_transpose_b() ? *b_shape.rbegin() : *++b_shape.rbegin();
-            const size_t brgemm_vnni_factor_for_real16 = 2;  // 4/2(size in term of byte for bf16/fp16)
-            if (is_bf16 || is_fp16)
-                return K.is_static() && (K.get_length() % brgemm_vnni_factor_for_real16 == 0);
-            if (is_int8)
-                return K.is_static();
-        }
+        if (is_fp32)
+            return dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx2);
         if (is_int8)
-            return dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_vnni) ||
+            return dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_amx) ||
+                   dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_vnni) ||
```

Comment:
Should include `avx2_vnni_2` as well? We handle this case in `get_primitive_isa` [here](https://github.com/openvinotoolkit/openvino/blob/8f0094dabda2dfe02c8414fd13f7d268c06ce6c7/src/plugins/intel_cpu/src/transformations/snippets/x64/op/brgemm_utils.cpp#L48)

Reply:
As far as I understand, if machine has `avx2_vnni_2`, the call `mayiuse(avx2_vnni)` [return true](https://github.com/openvinotoolkit/oneDNN/blob/03ea468815d37b644a4557eb8b4a77a75adb9835/src/cpu/x64/cpu_isa_traits.hpp#L394) too. So I think that check for `avx2_vnni_2` is important only in `get_primitive_isa` to correct kernel creation, while `mayiuse(avx2_vnni)` is enough in plugin callback.

Thanks!